cpp/src/arrow/dataset/discovery.h

```diff
@@ -165,15 +165,17 @@ struct FileSystemFactoryOptions {
   // (resulting in an error at scan time).
   bool exclude_invalid_files = false;
 
-  // Files matching one of the following prefix will be ignored by the
-  // discovery process. This is matched to the basename of a path.
+  // Files and directories matching one of the following prefixes will be ignored
```

Comment:
I'd make `When using a selector (and not to an explicit list of files),` the first phrase so it doesn't catch off-guard those who don't read until the end.

Maybe rename the property name, e.g. `selector_ignore_prefixes`?

Reply:
Good call, will reword